targeteditor/src/main/scala/explore/targeteditor/TargetEditor.scala

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.target
+
+import cats.implicits._
+import explore.components.graphql.SubscriptionRenderMod
+import explore.implicits._
+import explore.model.SiderealTarget
+import explore.target.TargetQueries._
+import gem.Observation
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.html_<^._
+import japgolly.scalajs.react.vdom.html_<^._
+import monocle.function.Cons.headOption
+import react.common._
+import react.common._
+
+final case class TargetEditor(
+  observationId:    CtxIO[Observation.Id]
+)(implicit val ctx: AppContextIO)
+    extends ReactProps {
+  @inline override def render: VdomElement = TargetEditor.component(this)
+}
+
+object TargetEditor    {
+  type Props = TargetEditor
+
+  val component =
+    ScalaComponent
+      .builder[Props]("TargetEditor")
+      .render_P { props =>
+        implicit val appCtx = props.ctx
```

Comment:
With the latest crystal, this can now be rewritten as `props.observationId.withCtx { implicit appCtx => ... }`